services/thumbnails/README.md

```diff
@@ -39,6 +39,21 @@ If a file type was not properly assigned or the type identification failed, thum
 
 Thumbnails can either be generated as `png`, `jpg` or `gif` files. These types are hardcoded and no other types can be requested. A requestor, like another service or a client, can request one of the available types to be generated. If more than one type is required, each type must be requested individually.
 
+## Thumbnail Query String Parameters
+
+Clients can request thumbnail previews for files by adding `?preview=1` to the file URL. Requests for files with thumbnail availabe respond with HTTP status `404`.
+
+The following query parameters are supported:
+
+| Parameter | Required | Default Value                                        | Description                   |
+| --------- | -------- | ---------------------------------------------------- | ----------------------------- |
+| preview   | YES      | 1                                                    | generates preview             |
+| x         | YES      | first x-value configured in `THUMBNAILS_RESOLUTIONS` | horizontal target size        |
+| y         | YES      | first y-value configured in `THUMBNAILS_RESOLUTIONS` | vertical target size          |
+| a         | NO       | 1                                                    | aspect ratio                  |
+| c         | NO       | ???                                                  | ???                           |
```

Comment:
@kulmann what's the purpose of this parameter?

Reply:
The value is the `eTag` without the double quotes. Afaik the thumbnail service uses that for its caching strategy. But I don't know if and why we need to provide it as a client.

Reply:
I can see how it can help the server-side thumbnail provider save an internal lookup.

With `eTag`:
- server checks thumbnail storage for `fileID`+`eTag`+`size` combination:
    - if found: returns thumbnail
    - if not found: fetches file, generates thumbnail, stores and returns it
    
Without `eTag`:
- server fetches `eTag` for `fileID` (possibly expensive?)
- server checks thumbnail storage for `fileID`+`eTag`+`size` combination:
    - if found: returns thumbnail
    - if not found: fetches file, generates thumbnail, stores and returns it

Also, using the `eTag` should ensure that a proxy server inbetween doesn't return an old version from its cache but makes a new request to the underlying server, returning the correct version.